### PR TITLE
Update README_RELEASE_PROVIDER_PACKAGES.md

### PR DESCRIPTION
**Pull request by eladkal**
Update release guide after https://github.com/apache/airflow/pull/34253

### CODE REVIEW

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 1]
```diff
@@ -414,6 +414,16 @@ If you have providers as list of provider ids because you just released them you
...
 
+breeze release-management publish-docs providers-index amazon cncf.kubernetes --override-versioned --run-in-parallel
+
+breeze release-management add-back-references all-providers
+```
+
+or with
+
+```shell script
+cd "${AIRFLOW_REPO_ROOT}"
```
Comment (amoghrajesh) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md:425]: Is this missing some command later? It ends after CD, is this intentional?

Comment (eladkal) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md:425]: It doesn't end with cd the next command is executing a script
Open the file in the new version to see it.

**Review by amoghrajesh — APPROVED** [review]
Just checked. It looks awesome now

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 2]
```diff
@@ -414,6 +414,16 @@ If you have providers as list of provider ids because you just released them you
 ```shell script
 cd "${AIRFLOW_REPO_ROOT}"
 
+breeze release-management publish-docs providers-index amazon cncf.kubernetes --override-versioned --run-in-parallel
+
+breeze release-management add-back-references all-providers
```
Comment (eladkal) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md]: mmm I guess this is not right here?
in this part we want to release only some providers
so I guess I should change this to

`breeze release-management add-back-references amazon cncf.kubernetes`

to followup the above publish-docs one?

Comment (amoghrajesh) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md]: Oh yes on careful observation it should be what you mentioned. But I thought it was a general template. Let us make it the way you said

Comment (eladkal) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md]: see https://github.com/apache/airflow/pull/34295/commits/760681f03a947656f8faf2440d2b0e1176e40a33

Comment (amoghrajesh) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md]: Changes are fine but did we get the short hand options working for add back references command? I'm wondering that right now. Afk, or would've checked

**Review by hussein-awala — APPROVED** [review]

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 3]
```diff
@@ -414,6 +414,16 @@ If you have providers as list of provider ids because you just released them you
...
+breeze release-management publish-docs providers-index amazon cncf.kubernetes --override-versioned --run-in-parallel
+
+breeze release-management add-back-references providers-index amazon cncf.kubernetes
+```
+
+or with
+
+```shell script
+cd "${AIRFLOW_REPO_ROOT}"
+
```
Comment (potiuk) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md:426]: Should we remove the "publish_provider_documentation.sh" and refereence to it now?

Comment (eladkal) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md:426]: for now lets leave it as an option.
I will do another cleanup on the whole doc soon

**Pull request merged**